src/display/ui_state.rs

```diff
@@ -51,38 +51,61 @@ pub struct UIState {
 }
 
 impl UIState {
+    fn get_proc_name<'a>(
+        connections_to_procs: &'a HashMap<LocalSocket, String>,
+        local_socket: &LocalSocket,
+    ) -> Option<&'a String> {
+        if let Some(process_name) = connections_to_procs.get(local_socket) {
+            Some(process_name)
+        } else if let Some(process_name) = connections_to_procs.get(&LocalSocket {
```

Comment:
Out of curiosity, why do we need this special case?

Reply:
Because sometimes `lsof` outputs lines like
```
proc_name 74868            root   12u  IPv4 0xeb179a6646cbc645      0t0    UDP *:1234
```
where the local ip isn't revealed. In this case, `lsof_util` will insert a wildcard entry with IP 0.0.0.0 for v4(or ::0 for v6). This logic is attempt to see if there's any wildcard entry that matches the same port number.